kea-vscode/src/views/pull-request-contents/local-commit/local-folder-tree-node.ts

```diff
@@ -0,0 +1,31 @@
+import * as vscode from "vscode";
+import { CollapsibleState, getCollapsibleState, IParentTreeNode } from "../../tree-node";
+import { LocalFileTreeNode } from "./local-file-tree-node";
+
+/**
+ * Represents a folder in a local commit tree.
+ */
+export class LocalFolderTreeNode implements IParentTreeNode<LocalFileTreeNode | LocalFolderTreeNode> {
+  #contextValue = "localFolder";
+  #iconPath = new vscode.ThemeIcon("folder");
+  #tooltip = "Folder";
+  folderName: string;
+
+  collapsibleState: CollapsibleState = "collapsed";
+  children: Array<LocalFileTreeNode | LocalFolderTreeNode>;
+
+  constructor(folderPath: string) {
+    this.folderName = folderPath.split("/").pop() ?? folderPath;
+    this.children = [];
+  }
+
+  getTreeItem = (): vscode.TreeItem => {
+    const treeItem = new vscode.TreeItem(this.folderName, getCollapsibleState(this.collapsibleState));
+    treeItem.contextValue = this.#contextValue;
+    treeItem.iconPath = this.#iconPath;
+    treeItem.tooltip = this.#tooltip;
+    return treeItem;
+  };
```

Comment:
Clicking a folder tree node results in the following log:

```log
2025-07-20 16:33:50.866 [error] Failed to parse decoration payload Failed to parse decoration payload
```

Reply:
This error persists. Please inspect all the folder tree nodes while investigating this.